src/cplscheme/tests/SerialImplicitCouplingSchemeTest.cpp

```diff
@@ -736,8 +736,8 @@ BOOST_AUTO_TEST_CASE(FirstOrderWithAcceleration)
     // first order extrapolation
     BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 4); // = 2*3 - 2
   } else if (context.isNamed(second)) {
-    // extrapolation only applied to accelerated data. So data written by first participant.
-    BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 3);
+    // first order extrapolation
+    BOOST_TEST(mesh->data(receiveDataIndex)->values()(0) == 5); // = 2*3 - 1
```

Comment:
This is an important change in the test which might make this PR breaking: Extrapolation is now applied to all data, not only to the accelerated data. This simplifies the implementation and this change is also not observable with our integration tests. Performing the extrapolation is also not expensive.